lib/core-integration/src/Test/Integration/Scenario/API/Shared/Wallets.hs

```diff
@@ -214,6 +215,83 @@ spec = describe "SHARED_WALLETS" $ do
+                "account_public_key": #{accXPubTxt},
+                "account_index": "30H",
+                "payment_script_template":
+                    { "cosigners":
+                        { "cosigner#0": #{accXPubTxt} },
+                      "template":
+                          { "all":
+                             [ "cosigner#0",
+                               { "active_from": 120 }
+                             ]
+                          }
+                    }
+                } |]
+        rPostAcctOther <- postSharedWallet ctx Default payloadAcctOther
+        verify (fmap (view #wallet) <$> rPostAcctOther)
+            [ expectResponseCode HTTP.status201 ]
+        let walAcctOther = getFromResponse id rPostAcctOther
+        let (ApiSharedWallet (Right walAcctOtherActive)) = walAcctOther
+
+        (walAcctOtherActive ^. #id)  `shouldNotBe` (walActive ^. #id)
+
+        let payloadAcctSame = Json [json| {
+                "name": "Shared Wallet",
+                "account_public_key": #{accXPubDerived},
+                "account_index": "30H",
+                "payment_script_template":
+                    { "cosigners":
+                        { "cosigner#0": #{accXPubDerived} },
+                      "template":
+                          { "all":
+                             [ "cosigner#0",
+                               { "active_from": 120 }
+                             ]
+                          }
+                    }
+                } |]
+        rPostAcctSame <- postSharedWallet ctx Default payloadAcctSame
+        verify (fmap (view #wallet) <$> rPostAcctSame)
+            [ expectResponseCode HTTP.status201 ]
+        let walAcctSame = getFromResponse id rPostAcctSame
+        let (ApiSharedWallet (Right walAcctSameActive)) = walAcctSame
+
+        (walAcctSameActive ^. #id)  `shouldBe` (walActive ^. #id)
+
+        let payloadAcctSameOtherScript = Json [json| {
+                "name": "Shared Wallet",
+                "account_public_key": #{accXPubDerived},
+                "account_index": "30H",
+                "payment_script_template":
+                    { "cosigners":
+                        { "cosigner#0": #{accXPubDerived} },
+                      "template":
+                          { "all":
+                             [ "cosigner#0",
+                               { "active_from": 100 }
+                             ]
+                          }
+                    }
+                } |]
+        rPostAcctSameOtherScript <- postSharedWallet ctx Default payloadAcctSameOtherScript
+        verify (fmap (view #wallet) <$> rPostAcctSameOtherScript)
+            [ expectResponseCode HTTP.status201 ]
+        let walAcctSameOtherScript = getFromResponse id rPostAcctSameOtherScript
+        let (ApiSharedWallet (Right walAcctSameOtherScriptActive)) = walAcctSameOtherScript
+
+        (walAcctSameOtherScriptActive ^. #id)  `shouldNotBe` (walActive ^. #id)
```

Comment:
This test became quite lengthy and not easy to follow... If I'm reading it correctly the ids are not the same because of different templates (active_from:120) vs. (active_from:100)?

Maybe it would make sense to separate this test into two separate ones? One verifying the case with the same ids and the other verifying the case with different ids...

Reply:
indeed, the same accXPub but difference in any script should give rise to different wallet id. I wanted to add those expectations under already present test to avoid consuming faucet. But we can separate it :+1:

Reply:
Those tests should not consume faucet as no coins are involved.

Reply:
ok, Piotr, constructed dedicated test just for you ;-)